src/components/Collapse.tsx

```diff
@@ -1,37 +1,40 @@
-import clsx from "clsx";
-import { createEffect, createSignal, JSX, onCleanup, onMount, splitProps } from "solid-js";
+import clsx from "clsx"
+import { createEffect, createSignal, JSX, onCleanup, onMount, splitProps } from "solid-js"
 
 export default function Collapse(
 	props: { children: JSX.Element; isOpen: boolean } & JSX.IntrinsicElements["div"],
 ) {
 	const [, divProps] = splitProps(props, ["children", "isOpen"])
+	const [contentHeight, setContentHeight] = createSignal()
 
-	const inner = (<div>{props.children}</div>) as HTMLElement
+	let outer: HTMLElement | undefined
+	let inner: HTMLElement | undefined
 
-	const [contentHeight, setContentHeight] = createSignal(inner.clientHeight)
-
-	const outer = (
-		<div {...divProps} class={clsx("relative overflow-hidden transition-[height]", divProps.class)}>
-			{inner}
-		</div>
-	) as HTMLElement
-
-	const ro = new ResizeObserver(([entry]) => {
-		setContentHeight(entry.contentRect.height)
-	})
-	ro.observe(inner)
-
-	onCleanup(() => {
-		ro.disconnect()
+	createEffect(() => {
+		const ro = new ResizeObserver(([entry]) => {
+			setContentHeight(entry.contentRect.height)
+		})
+		ro.observe(inner!)
+		onCleanup(() => {
+			ro.disconnect()
+		})
 	})
 
 	onMount(() => {
-		inner.style.position = "absolute"
+		inner!.style.position = "absolute"
```

Comment:
nit: safer to wrap this in an `if(inner)` just in case something crazy happens and the node wasn't found (same below)

Reply:
I actually don't think so... if the node isn't found, it would break silently, and on an error, at least we're aware that a mistake is made, and it doesn't take the entire site down, just this callback fails to run 😅

Reply:
Ah ha, I was just guessing it'd be unhandled and break JS.  Just tried it locally and the Collapse throws an error but doesn't actually break anything else :tada: